### PR TITLE
typical 0.9.5 (new formula)

### PR DESCRIPTION
**Pull request by stepchowfun**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by stepchowfun** [comment]
Note: `brew audit --new typical` does return errors, but they seem unrelated to the new formula:

```sh
$ brew audit --new typical
homebrew/core
  * audit_exceptions/versioned_dependencies_conflicts_allowlist.json references
    formulae or casks that are not found in the homebrew/core tap.
    Invalid formulae or casks: predictionio
  * pypi_formula_mappings.json references
    formulae or casks that are not found in the homebrew/core tap.
    Invalid formulae or casks: rtv
Error: 2 problems in 1 tap detected.
```

**Review by chenrui333 — DISMISSED** [review]

**Comment by chenrui333** [comment]
> Note: `brew audit --new typical` does return errors, but they seem unrelated to the new formula:
> 
> ```shell
> $ brew audit --new typical
> homebrew/core
>   * audit_exceptions/versioned_dependencies_conflicts_allowlist.json references
>     formulae or casks that are not found in the homebrew/core tap.
>     Invalid formulae or casks: predictionio
>   * pypi_formula_mappings.json references
>     formulae or casks that are not found in the homebrew/core tap.
>     Invalid formulae or casks: rtv
> Error: 2 problems in 1 tap detected.
> ```

do you actually have homebrew-core branch update to date?

**Comment by chenrui333** [comment]
the new formula audit passed in my local as well as CI

**Comment by stepchowfun** [comment]
Ah indeed, my copy of `Homebrew/homebrew-core` was not up to date, but my copy of `Homebrew/brew` was. I must have been confused by the nested repositories. It passes for me now as well after updating.

File: Formula/typical.rb [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+          body: String = 2
+      }
+
+      choice SendEmailResponse {
+          success = 0
+          error: String = 1
+      }
+    EOS
+
+    assert_match "", shell_output("#{bin}/typical generate types.t --rust types.rs --typescript types.ts")
```
Comment (chenrui333) [inline Formula/typical.rb:28]: actually we should do this
```suggestion
    assert_empty shell_output("#{bin}/typical generate types.t --rust types.rs --typescript types.ts")
```

File: Formula/typical.rb [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+    generated_rust_code = File.read(testpath/"types.rs")
+    generated_typescript_code = File.read(testpath/"types.ts")
```
Comment (chenrui333) [inline Formula/typical.rb:31]: maybe do this?
```suggestion
    generated_rust_code = (testpath/"types.rs").read
    generated_typescript_code = (testpath/"types.ts").read
```

**Comment by chenrui333** [comment]
actually left some notes to improve the test.

**Comment by stepchowfun** [comment]
Thanks for the suggestions! Applied.

**Review by chenrui333 — DISMISSED** [review]

**Pull request closed without merging**